STARK: Fix bug #14482 Texture mods with unusual names not loading

gordonfreeman01 opened the pull request:
STARK: Fix bug #14482 Texture mods with unusual names not loading

Addresses the following bug by re-encoding the special character causing it:
https://bugs.scummvm.org/ticket/14482

bluegr commented on engines/stark/resources/textureset.cpp:
```diff
@@ -117,7 +117,7 @@ void TextureSet::extractArchive() {
 
 Gfx::TextureSet *TextureSet::readOverrideDdsArchive() {
-	Common::String archiveName = _filename + ".zip";
+Common::String archiveName = _filename + ".zip";
```
Why? Please revert this

gordonfreeman01 replied:
Huh, this is weird, I didn't actually modify that myself (doesn't show up as a diff in VS either). Any idea what might be causing this?

lephilousophe commented on engines/stark/resources/textureset.cpp:
```diff
@@ -169,6 +171,15 @@ Gfx::TextureSet *TextureSet::readOverrideDdsArchive() {
+		//	\x91 is ae in CP437 and CP850 which is used by some compression utilities
+		//	\xc3\xa6 is the UTF-8 2-byte representation
+		// Both get converted to the game's original encoding here
+		if (textureName == "pupp\x91""r" || textureName == "pupp\xc3\xa6""r") {
```
Why are you doubling the quote here?
It doesn't seem to be needed as per below.
Either you separate the "r" in every places either you don't.

gordonfreeman01 replied:
> Why are you doubling the quote here? It doesn't seem to be needed as per below. Either you separate the "r" in every places either you don't.

I implemented feedback from @sev- , personally I think it would be more readable to have it in one block but will go with the project consensus. Would a single set of quotes be preferred?

aquadran replied:
single quote is preferred

gordonfreeman01 replied:
Ok cool, that's done in the latest version of the branch.

gordonfreeman01 replied:
Fixed in new PR https://github.com/scummvm/scummvm/pull/5251 as Github closed this one after I rebased my branch.

gordonfreeman01 commented:
@bluegr @lephilousophe I attempted some updates based on your feedback and rebased my branch but it ended up closing the PR automatically as a result, apologies. Is there any other process used to update the branch from master and making a PR? I was pulling from master then rebasing my changes on that updated mirror of master for this PR but any rebase seems to end the PR and merge commits are against wiki instructions.

Concerning the issue raised by @bluegr I didn't edit the tabs in the and the encoding seems to match with other source files, any idea what might be causing the tab to be removed during push?

gordonfreeman01 commented:
> @bluegr @lephilousophe I attempted some updates based on your feedback and rebased my branch but it ended up closing the PR automatically as a result, apologies. Is there any other process used to update the branch from master and making a PR? I was pulling from master then rebasing my changes on that updated mirror of master for this PR but any rebase seems to end the PR and merge commits are against wiki instructions.
> 
> Concerning the issue raised by @bluegr I didn't edit the tabs in the and the encoding seems to match with other source files, any idea what might be causing the tab to be removed during push?

Please ignore, the tab issue was just a VS setting to ignore whitespace diffs mistakenly left on. For anyone else reading this, I created a new https://github.com/scummvm/scummvm/pull/5251 as Github closed this one after I rebased my branch, individual threads have been updated with the link to the new PR.
